sdk/Pulumi/Provider/ComponentResourceProviderBase.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Pulumi.Utilities;
+
+namespace Pulumi.Experimental.Provider;
+
+public class ComponentResourceProviderBase : Provider
```

Comment:
I think this could just get merged into Provider. There are cases of providers that support Custom and Component resources, there's no need to make components require a different base class for that.